pkg/cluster/platformworkloadidentities.go

```diff
@@ -20,24 +20,23 @@ func (m *manager) platformWorkloadIdentityIDs(ctx context.Context) error {
 	}
 
 	identities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
-	updatedIdentities := make([]api.PlatformWorkloadIdentity, len(identities))
+	updatedIdentities := make(map[string]api.PlatformWorkloadIdentity, len(identities))
 
 	for i, identity := range identities {
 		resourceId, err := arm.ParseResourceID(identity.ResourceID)
 		if err != nil {
-			return fmt.Errorf("platform workload identity '%s' invalid: %w", identity.OperatorName, err)
+			return fmt.Errorf("platform workload identity '%s' invalid: %w", i, err)
```

Comment:
Nit: I'd rename the loop variable from `i` to something that identifies that it's the name of the operator. Otherwise it's easy to read this and get confused thinking that `i` is a number.